webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Modal, ModalVariant, FormSelect,
+  FormSelectOption, Checkbox, Form, FormGroup,
+  ActionGroup, Button } from '@patternfly/react-core';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import { selectCVDetails,
+  selectCVDetailStatus,
+  selectCVDetailError } from '../../Details/ContentViewDetailSelectors';
+import { addComponent } from '../ContentViewDetailActions';
+
+const ComponentContentViewAddModal = ({
+  cvId, componentCvId, componentId, latest, show, setIsOpen,
+}) => {
+  const dispatch = useDispatch();
+  const componentDetails = useSelector(
+    state => selectCVDetails(state, componentCvId),
+    shallowEqual,
+  );
+  const componentStatus = useSelector(
+    state => selectCVDetailStatus(state, componentCvId),
+    shallowEqual,
+  );
+  const componentError = useSelector(
+    state => selectCVDetailError(state, componentCvId),
+    shallowEqual,
+  );
+  const [cvName, setCvName] = useState('');
+  const [options, setOptions] = useState([]);
+  const [useLatest, setUseLatest] = useState(componentId ? latest : false);
```

Comment:
Should probably call this something that doesn't start with `use`, since it's not a custom hook.

Reply:
> one more thought I just had: When you check the Automatically update to latest version box, the dropdown is disabled. Which is good. But if you happen to have a non-latest version selected, it implies that it will be set to the old version and yet automatically update, which is not the case. So, when the checkbox is checked, I think the dropdown should disable _and_ set itself to the latest version.

This makes sense to me.